src/main/java/core/basesyntax/FileWork.java

```diff
@@ -8,7 +14,31 @@
  * Результат: web wide width world</p>
  */
 public class FileWork {
+    public static final String FIRST_CHAR = "w";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        String[] result = new String[0];
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            ArrayList searchedWords = new ArrayList();
+            while ((line = bufferedReader.readLine()) != null) {
+                String[] bufferedWords = line.toLowerCase()
+                        .replaceAll("[^a-z]", " ")
+                        .split(" ");
+                for (String word: bufferedWords) {
+                    if (word.startsWith(FIRST_CHAR)) {
+                        searchedWords.add(word);
+                    }
+                }
+            }
+            result = new String[searchedWords.size()];
```

Comment:
https://www.geeksforgeeks.org/arraylist-array-conversion-java-toarray-methods/

Reply:
toArray(), in this case, works only if we cast ArrayList to String with generic... I'm trying to avoid using generic as much as I can, cause it's our next lesson... but OK, I surrender :-p